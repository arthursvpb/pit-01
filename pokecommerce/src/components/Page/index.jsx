import React from "react";
import ClayLayout from "@clayui/layout";

export default function index({ title, children }) {
  return (
    <ClayLayout.Container className="mt-4">
      <h1>{title}</h1>

      <div>{children}</div>
    </ClayLayout.Container>
  );
}
