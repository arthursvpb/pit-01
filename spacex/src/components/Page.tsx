import React from "react";
import { Box, Heading, Container, Button } from "react-bulma-components";
import Loading from "../components/Loading";

type PageProps = {
  title: string;
  description: string;
  onClickBack?: () => void;
  loading?: boolean;
};

const Page: React.FC<PageProps> = ({
  title,
  description,
  onClickBack,
  loading,
  children,
}) => {
  return (
    <Container style={{ marginTop: 40 }}>
      <Box>
        <Heading>{title}</Heading>
        <Heading subtitle size={6}>
          {description}
        </Heading>

        {onClickBack && <Button onClick={onClickBack}>Back</Button>}
      </Box>

      {loading ? <Loading /> : children}
    </Container>
  );
};

export default Page;
