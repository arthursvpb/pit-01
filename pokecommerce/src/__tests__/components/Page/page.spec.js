import Page from "../../../components/Page";

import { render } from "@testing-library/react";

describe("Page Component", () => {
  it("renders", () => {
    const { asFragment } = render(<Page />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("render with props", () => {
    const { queryByText } = render(
      <Page title="Pokemon List">
        <div className="pokemon--list">
          <span>Pokemon A</span>
          <span>Pokemon B</span>
          <span>Pokemon C</span>
        </div>
      </Page>
    );

    expect(queryByText("Pokemon List")).toBeTruthy();
    expect(queryByText("Pokemon A")).toBeTruthy();
    expect(queryByText("Pokemon B")).toBeTruthy();
    expect(queryByText("Pokemon C")).toBeTruthy();
  });
});
