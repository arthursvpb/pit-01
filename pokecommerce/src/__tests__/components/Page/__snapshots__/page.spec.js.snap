// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Component renders 1`] = `
<DocumentFragment>
  <div
    class="mt-4 container"
  >
    <h1 />
    <div />
  </div>
</DocumentFragment>
`;
